### ISSUE TITLE
bug: errors pushing to ECR on AWS build

### ISSUE THREAD

**Issue by qdzlug** [nginxinc/kic-reference-architectures#134]
**Describe the bug**
When running in AWS, the code fails with errors:

```
Previewing update (jaynew01)

View Live: https://app.pulumi.com/qdzlug/kic-image-push/jaynew01/previews/1672830f-f91d-49ac-81cd-18a90724ef90

     Type                               Name                                Plan       
 +   pulumi:pulumi:Stack                kic-image-push-jaynew01             create     
 +   └─ pulumi-python:dynamic:Resource  ingress-controller-repository-push  create     
 
Resources:
    + 2 to create

Updating (jaynew01)

View Live: https://app.pulumi.com/qdzlug/kic-image-push/jaynew01/updates/1

     Type                               Name                                Status      Info
 +   pulumi:pulumi:Stack                kic-image-push-jaynew01             created     7 messages
 +   └─ pulumi-python:dynamic:Resource  ingress-controller-repository-push  created     
 
Diagnostics:
  pulumi:pulumi:Stack (kic-image-push-jaynew01):
    /home/jschmidt/data/repos/kic-reference-architectures/pulumi/python/utility/kic-image-push/repository_push.py:83: RuntimeWarning: coroutine 'login_to_registry' was never awaited
      docker.login_to_registry(registry=registry, log_resource=self.resource)
    RuntimeWarning: Enable tracemalloc to get the object allocation traceback
    /home/jschmidt/data/repos/kic-reference-architectures/pulumi/python/utility/kic-image-push/repository_push.py:90: RuntimeWarning: coroutine 'tag_and_push_image' was never awaited
      docker.tag_and_push_image(repository_url=repository_url,
    RuntimeWarning: Enable tracemalloc to get the object allocation traceback
    info: Tagged and pushed image [nginx/nginx-ingress:2.2.0] to [369313531325.dkr.ecr.us-west-2.amazonaws.com/ingress-controller-jaynew01:2.2.0]
 
Outputs:
    ecr_repository: {
        ecr_image_id        : "sha256:5f3cd9092a9002e5d73af22ba1737ceb0ee491e457ad55e49ed6b0ddec8849a4"
        ecr_image_name      : "369313531325.dkr.ecr.us-west-2.amazonaws.com/ingress-controller-jaynew01:2.2.0"
        id                  : "7834ac50-3178-49e2-a918-dd6525b770d2"
        image_id            : "sha256:5f3cd9092a9002e5d73af22ba1737ceb0ee491e457ad55e49ed6b0ddec8849a4"
        image_name          : "nginx/nginx-ingress:2.2.0"
        image_tag           : "2.2.0"
        repository_password : [secret]
        repository_url      : "369313531325.dkr.ecr.us-west-2.amazonaws.com/ingress-controller-jaynew01"
        repository_username : [secret]
        urn                 : "urn:pulumi:jaynew01::kic-image-push::pulumi-python:dynamic:Resource::ingress-controller-repository-push"
    }

Resources:
    + 2 created

Duration: 4s

```
This leads to a failure when we try and deploy the IC; when we dig into this further we see that no image is actually pushed to the registry.

**To Reproduce**

Deploy MARA to AWS. This current happens consistently on the `jaymistake02` branch; will be checking with main as well.

**Expected behavior**
Image should be pushed and then pulled by the IC deployment.

**Your environment**
* n/a

**Additional context**
None

**Comment by qdzlug** [comment]
Work-around is to regress to 3.1.0 for pulumi-docker.

**Comment by qdzlug** [comment]
Closing; this has been fixed in the 1.1 release and will be further address when #167 merges.
